contracts/JumpBall.cdc

```diff
@@ -48,109 +55,178 @@ access(all) contract JumpBall {
+            }
             let nftID = nft.id
             self.nfts[nftID] <-! nft
             self.ownership[nftID] = owner
             emit NFTDeposited(gameID: self.id, nftID: nftID, owner: owner)
         }
 
-        // Award all NFTs to the winner
-        access(contract) fun transferAllToWinner(winnerCap: Capability<&{NonFungibleToken.Collection}>) {
-            let winnerCollection = winnerCap.borrow() ?? panic("Failed to borrow winner capability.")
-
-            for (nftID, nft) in self.nfts {
-                let previousOwner = self.ownership.remove(key: nftID)!
-                winnerCollection.deposit(token: <-nft)
-                emit NFTAwarded(gameID: self.id, nftID: nftID, previousOwner: previousOwner, winner: winnerCap.address)
+        // Return all NFTs to their original owners
+        access(contract) fun returnAllNFTs() {
+            let keys = self.nfts.keys
+            for key in keys {
+                let owner = self.ownership[key] ?? panic("Owner not found for NFT")
+                let depositCap = JumpBall.getDepositCapForAddress(owner: owner)
+                self.returnNFT(nftID: key, depositCap: depositCap)
             }
-
-            // Clear all NFTs after awarding them to the winner
-            self.nfts = {}
         }
 
-        // Return an NFT to its original owner
+        // Return a specific NFT to its original owner
         access(contract) fun returnNFT(nftID: UInt64, depositCap: Capability<&{NonFungibleToken.Collection}>) {
             pre {
                 self.ownership.containsKey(nftID): "NFT does not exist in this game."
             }
-
             let ownerAddress = self.ownership.remove(key: nftID)!
             let receiver = depositCap.borrow() ?? panic("Failed to borrow receiver capability.")
             receiver.deposit(token: <-self.nfts.remove(key: nftID)!)
             emit NFTReturned(gameID: self.id, nftID: nftID, owner: ownerAddress)
         }
 
-        destroy() {
-            destroy self.nfts
+        // Award all NFTs to the winner
+        access(contract) fun transferAllToWinner(winner: Address, winnerCap: Capability<&{NonFungibleToken.Collection}>) {
+            let winnerCollection = winnerCap.borrow() ?? panic("Failed to borrow winner capability.")
+            let keys = self.nfts.keys
+            for key in keys {
+                let nft <- self.nfts.remove(key: key) ?? panic("NFT not found.")
+                let previousOwner = self.ownership.remove(key: key)!
+                winnerCollection.deposit(token: <-nft)
+                emit NFTAwarded(gameID: self.id, nftID: key, previousOwner: previousOwner, winner: winner)
+            }
+        }
+    }
+
+    // Admin resource for game management
+    access(all) resource Admin {
+        access(all) fun determineWinner(gameID: UInt64, winnerCap: Capability<&{NonFungibleToken.Collection}>, stats: {UInt64: UInt64}) {
```

Comment:
should receiver cap for both player be stored as part of the game so that the cap can be used here instead of being able to pass any cap as function arg?